### PR TITLE
Add check for Captcha Form

### PR DESCRIPTION
**Pull request by mh4ck**
* Add check if captcha form exists on video page
* Throw an error with captcha form details

### CODE REVIEW

File: lib/info.js [hunk 1]
```diff
@@ -48,6 +48,18 @@ exports.getBasicInfo = async (id, options) => {
     }
   }
 
+  // Check if there is any captcha form
+  const captchaForm = util.between(body, '<form action="/das_captcha" method="POST">', '</form>');
+  if(captchaForm) {
```
Comment (fent) [inline lib/info.js:53]: lint: space before `{` please

File: lib/info.js [hunk 2]
```diff
@@ -48,6 +48,18 @@ exports.getBasicInfo = async (id, options) => {
...
+    let details = {};
+    details.sitekey = util.between(captchaForm, 'data-sitekey="', '"');
+    details.next = util.between(captchaForm, 'name="next" value="', '"');
+    details.session_token = util.between(captchaForm, 'name="session_token" value="', '"');
+    details.action_recaptcha_verify2 = util.between(captchaForm, 'name="action_recaptcha_verify2" value="', '"');
+    details.largeText = util.between(body, '<p class=\'largeText\'">', '</p>');
+    throw Error("Captcha form received with info: " + JSON.stringify(details));
```
Comment (fent) [inline lib/info.js:60]: is it better to print the json string of `details` here, or to attach all of these properties to the error object?

**Comment by mh4ck** [comment]
> thanks for doing this. left a couple of comments.
> 
> I've been working on a change that uses a different endpoint than the watchpage, not sure how this is going to work with it. it might be that captcha is no longer an issue with the other endpoint since it gives a json response.

Oh that would be very nice, so you think we don't need this check any longer?
Then just remove this PR ^^

**Comment by fent** [comment]
> Oh that would be very nice, so you think we don't need this check any longer?
> Then just remove this PR ^^

I don't know if it actually fixes it since I've never gotten the issue. the change is at
https://github.com/fent/node-ytdl-core/tree/new-endpoint

maybe you could try it and report if the captcha message is gone or displayed differently

**Comment by TimeForANinja** [comment]
> > Oh that would be very nice, so you think we don't need this check any longer?
> > Then just remove this PR ^^
> 
> I don't know if it actually fixes it since I've never gotten the issue. the change is at
> https://github.com/fent/node-ytdl-core/tree/new-endpoint
> 
> maybe you could try it and report if the captcha message is gone or displayed differently

Did check it, downloaded all uploads (means ~50 requests to youtube to resolve playlists) from NoCopyrightSounds (744 videos each ~3.5MB audio only) twice and only had two statusCode `403`. Used my original 25 workers in parallel script - ending with ~40 Video downloads a minute. No more captcha for me...

**Comment by fent** [comment]
superseded by https://github.com/fent/node-ytdl-core/pull/605

**Pull request closed without merging**